src/mapdata.h

```diff
@@ -196,6 +209,7 @@ struct map_data_common_t {
 
     map_bash_info        bash;
     map_deconstruct_info deconstruct;
+    map_dismantle_info   dismantle;
```

Comment:
That would still have the problem of the required tools being on the JSON side of things, and making required tools optional in certain contexts would be even more of a headache that this was to implement, and be harder to convey to the player.